streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -527,12 +527,10 @@ private void runLoop() {
             if (requiresPoll) {
                 requiresPoll = false;
 
-                boolean longPoll = totalNumBuffered == 0;
-
                 ConsumerRecords<byte[], byte[]> records = null;
 
                 try {
-                    records = consumer.poll(longPoll ? this.pollTimeMs : 0);
+                    records = consumer.poll(this.pollTimeMs);
```

Comment:
A side note here is that, inside consumer we usually issue another fetch request right after accumulated the responses from the previous request before returning the extracted records from the response; this is to enable pipelining the processing of the records while processing these records; I think with this pattern we still have such pipelining benefits but maybe less effective, here is my "guess":

1. pipelining will be most effective if 

```
the processing-time of the records from this response == the round-trip time for the next request
```

2. if the former is smaller than the latter we will see `consumer.poll()` not return immediately but blocked; if the former is larger than the latter we will see the buffer keep increasing until it cause the partitions to be paused;

3. with the previous pattern, if the processing time keep fluctuate (assuming the round-trip time for the same batch size keeps the same) we have a self-tuning mechanism here to pause when processing is slow / resume the partitions when processing is faster; with this pattern we do not have that "cushion" anymore. But I think in practice the fluctuation may not be very large (e.g. you happen to do very light computation for this 1000 records and happen to do some heavy lifting for the next 1000 records) but it is worth observing if there is any side-effects for some scenarios.